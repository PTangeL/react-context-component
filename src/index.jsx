export Context from './Context'
export withContext from './withContext'
