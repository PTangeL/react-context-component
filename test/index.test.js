import React from 'react'
import PropTypes from 'prop-types'
import renderer from 'react-test-renderer'
import { Context } from '../src'
import { withContext } from '../src'

const RenderDataContext = (props, { data }) => (
  <p>data from context: {data}, data from props {props.data}</p>
)

RenderDataContext.contextTypes = {
  data: PropTypes.number.isRequired
}

it('renders Context correctly', () => {
  const tree = renderer
    .create(
      <Context data={1}>
        <RenderDataContext />
      </Context>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a component withContext correctly', () => {
  const RenderWithDataContext = withContext({
    data: PropTypes.number.isRequired
  })(RenderDataContext)

  const tree = renderer
    .create(
      <Context data={1}>
        <RenderWithDataContext />
      </Context>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
