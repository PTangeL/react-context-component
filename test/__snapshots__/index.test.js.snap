// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Context correctly 1`] = `
<p>
  data from context: 
  1
  , data from props 
</p>
`;

exports[`renders a component withContext correctly 1`] = `
<p>
  data from context: 
  1
  , data from props 
  1
</p>
`;
